app/models/concerns/smooch_messages.rb

```diff
@@ -144,13 +144,22 @@ def send_message_if_disabled_and_return_state(uid, workflow, state)
       disabled ? 'disabled' : state
     end
 
+    # Used for incoming messages (e.g. message:appUser)
+    # where full message contents available
     def get_platform_from_message(message)
       type = message.dig('source', 'type')
       platform = type ? ::Bot::Smooch::SUPPORTED_INTEGRATION_NAMES[type].to_s : 'Unknown'
       RequestStore.store[:smooch_bot_platform] = platform
       platform
     end
 
+    # Used for outgoing messages (e.g. message:delivery:channel) where
+    # message contents are truncated
+    def get_platform_from_payload(payload)
```

Comment:
Question for review: does it make sense to have this method here? I thought of modifying the method above to allow either a payload or message JSON to be passed, but thought that it might be confusing and also that we might not want to alter the request store for sent messages, since we aren't already.

Reply:
Yes, I think it makes more sense to have it as a separate method.